hub/util/casting.py

```diff
@@ -37,6 +37,21 @@ def get_dtype(val: Union[np.ndarray, Sequence, Sample]) -> np.dtype:
         raise TypeError(f"Cannot infer numpy dtype for {val}")
 
 
+def get_htype(val: Union[np.ndarray, Sequence, Sample]) -> str:
+    if isinstance(val, np.ndarray):
+        return "generic"
+    types = set((map(type, val)))
+    if dict in types:
```

Comment:
Don't all types have to be dict?  Or, what's the case when not all are dicts (please add test case)?

Reply:
Test added.